via/views/route_by_content.py

```diff
@@ -49,9 +51,13 @@ def route_by_content(request):
         headers = {"Cache-Control": "no-cache"}
 
     via_url = request.registry.settings["legacy_via_url"]
-    url = request.path_qs.lstrip("/")
 
-    return exc.HTTPFound(f"{via_url}/{url}", headers=headers)
+    return exc.HTTPFound(
+        QueryParams.build_url(
+            f"{via_url}/{path_url}", request.params, strip_via_params=False
```

Comment:
```suggestion
            f"{via_url}/{path_url}", request.params
```

It defaults to `False`

Reply:
Yeah, I think I might remove the default to force you to say. This was to make crystal clear the intent.